lib/Semantics/resolve-names.cpp

```diff
@@ -5094,6 +5119,25 @@ void ConstructVisitor::Post(const parser::TypeGuardStmt::Guard &x) {
   }
 }
 
+void ConstructVisitor::Post(const parser::SelectRankCaseStmt::Rank &x) {
+  if (auto *symbol{MakeAssocEntity()}) {
+    SetTypeFromAssociation(*symbol);
+    SetAttrsFromAssociation(*symbol);
+    std::visit(
+        common::visitors{
+            [&](const parser::ScalarIntConstantExpr &init) {
+              Walk(init);
+              MaybeIntExpr expr{EvaluateIntExpr(init)};
+              if (auto val{evaluate::ToInt64(expr)}) {
+                SetRankFromParserNode(*symbol, *val);
+              }
```

Comment:
What happens if the expression is not a constant?  Is this the place where an error message should be emitted?

Should you validate the value of the `RANK` case here instead?

Reply:
Do you mean "not a constant" as other values of below variant?
`std::variant<ScalarIntConstantExpr, Star, Default> u;`
If so currently `Star` and `Default` rank cases values won't see any `rank:` in printing symbols and the respective details.
This is how it would look for 
`RANK DEFAULT` and `RANK(*)` test cases
`arr: AssocEntity type: INTEGER(4) expr:arr`

Is `rank: 1` a better value for `*` and `DEFAULT`?